tests/test_unit.py

```diff
@@ -44,5 +45,128 @@ def test_disambiguate_toc_name(self):
 
         self.assertEqual(yaml_want, yaml_got)
 
+
+    def test_extract_docstring_info(self):
+
+        ## Test for normal input
+        top_summary1_want = "\nSimple test for docstring.\n\n"
+        summary_info1_want = {
+            'variables': {
+                'arg1': {
+                    'var_type': 'int',
+                    'description': 'simple description.'
+                },
+                'arg2': {
+                    'var_type': 'str',
+                    'description': 'simple description for `arg2`.'
+                }
+            },
+            'returns': [
+                {
+                  'var_type': 'str', 
+                  'description': 'simple description for return value.'
+                }
+            ],
+            'exceptions': [
+                {
+                  'var_type': 'AttributeError', 
+                  'description': 'if `condition x`.'
+                }
+            ]
+        }
+
+        summary_info1_got = {
+            'variables': {},
+            'returns': [],
+            'exceptions': []
+        }
+
+        summary1 = """
+Simple test for docstring.
+
+Args: 
+    arg1(int): simple description.
+    arg2(str): simple description for `arg2`.
+
+Returns:
+    str: simple description for return value.
+
+Raises:
+    AttributeError: if `condition x`.
+"""
+
+        top_summary1_got = _extract_docstring_info(summary_info1_got, summary1, "")
+
+        self.assertEqual(top_summary1_got, top_summary1_want)
+        self.assertEqual(summary_info1_got, summary_info1_want)
+
+
+        ## Test for input coming in mixed format.
```

Comment:
Consider creating a separate test case for each `summary`. It's a bit hard to follow all of these as-is.

Reply:
done.